File: prime.rs
use std::env;
use std::f32;

fn is_prime(n:u32)->bool{
    if n%2==0 {
        return false;
    }
    let mut i = 3;
    while i<((n as f32).sqrt() as u32)+1{
        if n%i==0 {
            return false;
        }
        i+=2;
    }
    return true;
}

fn main(){
    let args: Vec<String> = env::args().collect();
    if args.len()<2 {
        println!("Requires one argument, remember target");
    }
    let mut sum = 0;
    let target=args[1].parse::<u32>().unwrap();

        for i in 2..target{
            if is_prime(i) {
                sum+=1;
            }
        }
    println!("{}", sum);
}

File: src/prime.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
)

func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n == 2 {
		return true
	}
	if n%2 == 0 {
		return false
	}
	for i := 3; i < int(math.Sqrt(float64(n)))+1; i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Requires one argument, remember target")
	}
	var sum int64 = 0
	var target, _ = strconv.ParseInt(os.Args[1], 10, 64)

	var target2 = int(target)

	for i := 1; i < target2; i++ {
		if isPrime(i) {
			sum++
		}
	}
	fmt.Println(sum)
}
